ORST/Assets/Scripts/Core/Interactions/DoorZone/DoorZone.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections;
+using System.Collections.Generic;
+using Oculus.Interaction;
+using ORST.Core.ModuleTasks;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace ORST.Core.Interactions
+{
+    public class DoorZone : MonoBehaviour {
+        private bool m_DoorsUnlocked;
+        private Grabbable m_DoorHandle;
+        private GameObject m_Player;
+
+        private void Start() {
+            m_DoorHandle = GetComponent<Grabbable>();
+        }
+
+        private void OnTriggerEnter(Collider other) {
+            if (other.CompareTag("Player")) {
+                m_Player = other.gameObject;
+                List<ModuleTask> remModuleTasks = ModuleTasksManager.Instance.GetRemainingTasks();
+                m_DoorsUnlocked = remModuleTasks.Count <= 0;
+                if (!m_DoorsUnlocked) {
+                    PopupManager.Instance.DisplayTasks(remModuleTasks);
+                    PopupManager.Instance.OpenPopup();
+                }
+                
+                DoorHandleMoved();
+            }
```

Comment:
What do you think about replacing this with an early return?

```suggestion
            if (!other.CompareTag("Player")) {
                return;
            }

            m_Player = other.gameObject;
            List<ModuleTask> remainingTasks = ModuleTasksManager.Instance.GetRemainingTasks();
            m_DoorsUnlocked = remainingTasks.Count <= 0;
            if (!m_DoorsUnlocked) {
                PopupManager.Instance.DisplayTasks(remainingTasks);
                PopupManager.Instance.OpenPopup();
            }

            DoorHandleMoved();
```